### PR TITLE
[W5.11][W13-3]ChelseyOng

### PR DESCRIPTION
**Pull request by chel-seyy**
Added an 'Edit' command. Updated the user guide, test cases and unit tests.

### CODE REVIEW

**Comment by emobeany** [comment]
I am still picking up Java so there are still a lot of gaps in my knowledge. But good job following the coding standards! :) 

One suggestion (I have no idea how to) could be to allow users to edit only one detail rather than having to edit or type all the details again when calling the edit command. 

Good job nonetheless!

**Review by emobeany — CHANGES_REQUESTED** [review]
I am still picking up Java so there are still a lot of gaps in my knowledge. But good job following the coding standards! :)

One suggestion (I have no idea how to) could be to allow users to edit only one detail rather than having to edit or type all the details again when calling the edit command.

Good job nonetheless!

File: src/seedu/addressbook/commands/HelpCommand.java [hunk 1]
```diff
@@ -19,6 +19,7 @@ public CommandResult execute() {
                 AddCommand.MESSAGE_USAGE
                 + "\n" + DeleteCommand.MESSAGE_USAGE
                 + "\n" + ClearCommand.MESSAGE_USAGE
+                + "\n" + EditCommand.MESSAGE_USAGE
```
Comment (Sri-vatsa) [inline src/seedu/addressbook/commands/HelpCommand.java:22]: 👍

File: src/seedu/addressbook/parser/Parser.java [hunk 2]
```diff
@@ -11,17 +11,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (Sri-vatsa) [inline src/seedu/addressbook/parser/Parser.java]: It is always a good practice to only import the classes needed and not import all.

File: src/seedu/addressbook/parser/Parser.java [hunk 3]
```diff
@@ -137,6 +130,38 @@ private Command prepareAdd(String args) {
         }
     }
 
+    /**
+     * Parses arguments in the context of the edit person command.
```
Comment (Sri-vatsa) [inline src/seedu/addressbook/parser/Parser.java:145]: Commenting code is a good practice!

File: test/expected.txt [hunk 4]
```diff
@@ -264,29 +291,29 @@
...
 || Enter command: || [Command entered:  delete 1]
-|| Deleted Person: Adam Brown Phone: 111111 Email: adam@gmail.com Address: 111, alpha street Tags: 
+|| Deleted Person: Betsy Choo Phone: (private) 222222 Email: (private) benchoo@nus.edu.sg Address: (private) 222, beta street Tags: [secretive]
 || ===================================================
 || Enter command: || [Command entered:  list]
-|| 	1. Betsy Choo Tags: [secretive]
-|| 	2. Dickson Ee Phone: 444444 Address: 444, delta street Tags: [friends]
+|| 	1. Dickson Ee Phone: 444444 Address: 444, delta street Tags: [friends]
+|| 	2. Esther Potato Phone: 555555 Email: esther@not.a.real.potato Tags: [tubers][starchy]
 ||
```
Comment (Sri-vatsa) [inline test/expected.txt:317]: Great to see that you have added tests as well :)

File: test/java/seedu/addressbook/commands/EditCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        editCommand.setData(book, EMPTY_PERSON_LIST);
+        CommandResult result = editCommand.execute();
+        UniquePersonList people = book.getAllPersons();
+
+        assertEquals(1, people.immutableListView().size());
+        assertEquals(String.format(EditCommand.MESSAGE_SUCCESS, personToEdit), result.feedbackToUser);
+        assertTrue(people.contains(personToEdit));
+        assertFalse(people.contains(personToAdd));
+    }
+}
```
Comment (Sri-vatsa) [inline test/java/seedu/addressbook/commands/EditCommandTest.java:117]: Good attempt 👍

File: src/seedu/addressbook/parser/Parser.java [hunk 6]
```diff
@@ -11,7 +11,18 @@
...
+import seedu.addressbook.commands.Command;
+import seedu.addressbook.commands.DeleteCommand;
+import seedu.addressbook.commands.ExitCommand;
+import seedu.addressbook.commands.FindCommand;
+import seedu.addressbook.commands.HelpCommand;
+import seedu.addressbook.commands.IncorrectCommand;
+import seedu.addressbook.commands.ListCommand;
+import seedu.addressbook.commands.ViewAllCommand;
+import seedu.addressbook.commands.ViewCommand;
 import seedu.addressbook.data.exception.IllegalValueException;
```
Comment (Sri-vatsa) [inline src/seedu/addressbook/parser/Parser.java]: Great that you have noticed this coding violation and fixed it :)

File: docs/UserGuide.adoc [hunk 7]
```diff
@@ -151,6 +151,25 @@ Views all details of the 2nd person in the address book.
...
+be seen using the `viewall` command.
+
+Persons can have any number of tags (including 0).
+****
+
+Examples:
+
+* `edit John Doe p/98765432 e/johnd@hotmail.com a/John street, block 456, #01-01`
+* `edit Betsy Crowe pp/1234567 e/betsycrowe@hotmail.com pa/Newgate Prison t/criminal t/friend`
+
```
Comment (Sri-vatsa) [inline docs/UserGuide.adoc:172]: 👍

**Review by Sri-vatsa — APPROVED** [review]
👍 👍It is great to see that you have implemented a more challenging feature. Great work on writing JUnit tests, updating docs as well. 

Perhaps, you could think about how you can work with only editing only some fields, instead of editing all details each time. 

Please look through my comments and close the PR

**Comment by chel-seyy** [comment]
Noted thanks!

**Pull request closed without merging**